velox/expression/ComplexWriterTypes.h

```diff
@@ -1045,6 +1045,14 @@ class GenericWriter {
     return vector_->type();
   }
 
+  BaseVector* base() const {
+    return vector_;
```

Comment:
The purpose of exposing vector_ here is for allowing direct write to the base vector since we also directly read from GenericView. 

The vector pointer comes from `result` in `void extractAccumulators(char** groups, int32_t numGroups, VectorPtr* result)` and `void extractValues(char** groups, int32_t numGroups, VectorPtr* result)`
This is roughly where does the pointer come from.
```cpp
VectorWriter<typename FUNC::IntermediateType> writer;
auto vector = (*result)
                  ->as<typename TypeToFlatVector<
                      typename FUNC::IntermediateType>::type>();
vector->resize(numGroups);
writer.init(*vector);
```
```cpp
void init(vector_t& vector) {
  vector_ = &vector;
  writer_.initialize(vector_);
  if (vector.type()->isPrimitiveType()) {
    TypeKind kind = vector.typeKind();
    VELOX_DYNAMIC_TYPE_DISPATCH_ALL(ensureCastedWriter, kind);
  }
}
```

So the pointer will be valid as long as `GenericWriter` is valid. 

Tho I agree that we should use something like share_ptr would be better, but this requires a lot of changes from a lot files since we are basically change the design of `ComplexWriterType`

@kgpai Do you have recommended solutions?